BraintreeCore/src/main/java/com/braintreepayments/api/BraintreeShopperInsightsClient.kt

```diff
@@ -0,0 +1,41 @@
+package com.braintreepayments.api
+
+import androidx.annotation.VisibleForTesting
+
+/**
+ * Use `BTShopperInsightsClient` to optimize your checkout experience
+ * by prioritizing the customer’s preferred payment methods in your UI.
+ * By customizing each customer’s checkout experience,
+ * you can improve conversion, increase sales/repeat buys and boost user retention/loyalty.
+ * - Note: This feature is in beta. It's public API may change in future releases.
+ */
+class BraintreeShopperInsightsClient @VisibleForTesting internal constructor() {
+
+    /**
+     * Retrieves recommended payment methods based on the provided shopper insights request.
+     *
+     * This function takes a `ShopperInsightRequest` object as input and returns a `ShopperInsightResult`.
+     * The result includes flags indicating whether specific payment methods,
+     * such as PayPal or Venmo, are potentially a high priority for the shopper.
+     *
+     * @param request The `ShopperInsightRequest` containing information about the shopper.
+     * @return A `ShopperInsightResult` object indicating the recommended payment methods.
+     */
```

Comment:
```suggestion
     * Retrieves recommended payment methods based on the provided shopper insights request.
     *
     * @param request The [ShopperInsightRequest] containing information about the shopper.
     * @return A [ShopperInsightResult] object indicating the recommended payment methods.
     */
```

I don't think the extra explanation is necessary since we link to the classes where those details are documented, and we have param documentation. But open to whatever others think. Also I think [] syntax links the classes in kdoc.